tools/audio_common.py

```diff
@@ -0,0 +1,1193 @@
+		assert input.read(3) == b"AIF"
+		typeb = input.read(1)
+		if typeb == b"C":
+			self.is_aifc = True
+		
+		#Scan chunks and map
+		offset = 12
+		
+		while True:
+			magicno = input.read(4)
+			if not magicno or len(magicno) < 4:
+				break
+			secmagic = magicno.decode('utf-8')
+			secsize, = struct.unpack(">L", input.read(4))
+			offset += 8
+			if secmagic == "APPL":
+				self.appl_sections.append((offset, secsize))
+			else:
+				self.sections[secmagic] = (offset, secsize)
+			input.seek(offset + secsize)
+			offset += secsize
+			
+		input.close()
+		
+	def loadData(self, offset, size):
+		sec_dat = None
+		with open(self.path, 'rb') as f:
+			f.seek(offset)
+			sec_dat = f.read(size)
+		return sec_dat
+		
+	def loadSectionData(self, magicno):
+		section_loc = self.sections[magicno]
+		if section_loc is None:
+			return None
+		return self.loadData(section_loc[0], section_loc[1])
+		
+	def loadApplSectionData(self, idx):
+		if idx < 0 or idx >= len(self.appl_sections):
+			return None
+		section_loc = self.appl_sections[idx]
+		if section_loc is None:
+			return None
+		return self.loadData(section_loc[0], section_loc[1])
+			
+class AifcWriter:
+	def __init__(self, out):
+		self.out = out
+		self.sections = []
+		self.total_size = 0
+
+	def add_section(self, tp, data):
+		assert isinstance(tp, bytes)
+		assert isinstance(data, bytes)
+		self.sections.append((tp, data))
+		self.total_size += align(len(data), 2) + 8
+
+	def add_custom_section(self, tp, data):
+		self.add_section(b"APPL", b"stoc" + self.pstring(tp) + data)
+
+	def pstring(self, data):
+		return bytes([len(data)]) + data + (b"" if len(data) % 2 else b"\0")
+
+	def finish(self):
+		# total_size isn't used, and is regularly wrong. In particular, vadpcm_enc
+		# preserves the size of the input file...
+		self.total_size += 4
+		self.out.write(b"FORM" + struct.pack(">I", self.total_size) + b"AIFC")
+		for (tp, data) in self.sections:
+			self.out.write(tp + struct.pack(">I", len(data)))
+			self.out.write(data)
+			if len(data) % 2:
+				self.out.write(b"\0")
+
+#Util Functions 1
+def tryStr2Num(val):
+	try:
+		return int(val)
+	except:
+		try:
+			return float(val)
+		except:
+			return val
+
+def loadSoundData(aif_path):
+	aif_reader = AifReader(aif_path)
+	snd_data = aif_reader.loadSectionData('SSND')
+	
+	if snd_data is None:
+		#It SHOULD be SSND, but just in case, check for a CSND
+		snd_data = aif_reader.loadSectionData('CSND')
+		
+	return snd_data
+
+def toNote(note):
+	tone = (note + 9) % 12
+	octave = str(math.floor((((note + 9) / 12) + 1)))
+
+	tone_str = {
+		0: "C",
+		1: "C♯",
+		2: "D",
+		3: "D♯",
+		4: "E",
+		5: "F",
+		6: "F♯",
+		7: "G",
+		8: "G♯",
+		9: "A",
+		10: "A♯",
+		11: "B"
+	}.get(tone)
+
+	return "{0}{1}".format(tone_str, octave)
+
+def parseNoteName(noteName):
+	#This is crude but eh, just want an implementation for now
+	tone_str = noteName[0]
+	oct_str = noteName[1:]
+	if noteName[1] == '♯':
```

Comment:
I put this in instead and commented out my original. This won't work in cases where the octave string is > 1 character, but I don't THINK that should be a problem since octaves -1 and 10 should be out of range?

Reply:
They're _technically_ not, but nothing uses them in the game and no sane person would use them for a soundfont.